### PR TITLE
#286 Forbid imports across packages and apps

### LINKED ISSUE #286: Configure `eslint` to prevent imports between apps
We can configure `eslint` to prevent accidental imports from one app to another.
This will help us keep our codebase clean, builds fast and code reusable.

Common types and utilities should be moved to [packages/checkout-common](https://github.com/saleor/saleor-checkout/tree/main/packages/checkout-common) instead.

### PR DESCRIPTION
**Pull request by mmiszy**
Fixes #286

### CODE REVIEW

File: packages/checkout-common/src/utils.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export function assertUnreachable(value: never): never {
+  // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
   throw new Error(`Unexpected case: ${value}`);
```
Comment (witoszekdev) [inline packages/checkout-common/src/utils.ts:3]: Why can't we do this?
```suggestion
  throw new Error(`Unexpected case: ${value as string}`);
```

**Review by witoszekdev — COMMENTED** [review]
nice 👍🏻

File: packages/checkout-common/src/utils.ts (hunk 1 above)
Comment (mmiszy) [reply packages/checkout-common/src/utils.ts:3]: We can but why is that better? `value` is not a string, it could be anything.

Comment (witoszekdev) [reply packages/checkout-common/src/utils.ts:3]: You're right, that just avoids the problem, maybe we should provide a default if `value` is not a string - I think this ESLint rule makes sense, as we don't want to have `object [Object]` in an error 😅

**Review by witoszekdev — APPROVED** [review]

**Pull request merged**